spec/models/user_spec.rb

```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validation' do
+    it { is_expected.to validate_presence_of(:email) }
+    it { is_expected.to allow_value('valid@email.com').for(:email) }
+    it { is_expected.to_not allow_value('invalid.com').for(:email) }
```

Comment:
specs should describe all possible scenarios - and this is problem with the regex :) 

it's good to write smth like 

let(:valid_emails) { %w(user@example.com user12@example.com user_lorem@example12.pl ...) } 
let(:invalid_emails) { %w(invalid.com lorem@invalid lorem#@#%$$%$#$%@invalid.com ...) }

and do a loop specs :)